### PR TITLE
Payment user subscription info and payment flow

### PR DESCRIPTION
**Pull request by HancilSequeira**

### CODE REVIEW

File: catalog/controller/extension/payment/razorpay.php [hunk 1]
```diff
@@ -94,59 +126,124 @@ private function get_order_creation_data($order_id)
...
+
+    private function get_subscription_order_creation_data($order_id)
+    {
+        $this->load->model('extension/payment/razorpay');
+
+        $order = $this->model_checkout_order->getOrder($order_id);
+        $recurringPlanData = $this->cart->getProducts()[0]["recurring"];
+        $productId = $this->cart->getProducts()[0]['product_id'];
+
+        $planData = $this->model_extension_payment_razorpay->getPlanByRecurringIdAndFrequencyAndProductId($recurringPlanData['recurring_id'], $recurringPlanData['frequency'], $productId);
```
Comment (sofiya02) [inline catalog/controller/extension/payment/razorpay.php:152]: list of plan details from recurring table. can you map recurring table and plan id

File: catalog/controller/extension/payment/razorpay.php [hunk 2]
```diff
@@ -94,59 +126,124 @@ private function get_order_creation_data($order_id)
...
+
+        return ["subscriptionData" => $subscriptionData, "planData" => $planData];
+
+    }
+
 
     public function callback()
     {
         $this->load->model('checkout/order');
+        $this->load->model('extension/payment/razorpay');
```
Comment (sofiya02) [inline catalog/controller/extension/payment/razorpay.php:189]: unable to see razorpay PG at checkout

**Review by sofiya02 — CHANGES_REQUESTED** [review]
Overall reduce space and code format

File: catalog/controller/extension/payment/razorpay.php [hunk 3]
```diff
@@ -9,40 +9,76 @@ class ControllerExtensionPaymentRazorpay extends Controller
...
      */
-    const PAYMENT_AUTHORIZED    = 'payment.authorized';
-    const PAYMENT_FAILED        = 'payment.failed';
-    const ORDER_PAID            = 'order.paid';
+    const PAYMENT_AUTHORIZED = 'payment.authorized';
+    const PAYMENT_FAILED = 'payment.failed';
+    const ORDER_PAID = 'order.paid';
 
     // Set RZP plugin version
     private $version = '4.0.2';
```
[the remark this answers is not in the document]
Comment (HancilSequeira) [reply catalog/controller/extension/payment/razorpay.php:17]: Will be done once everything is finalized

File: catalog/view/theme/default/template/extension/payment/razorpay_subscription_info.twig [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+                <td class="text-left">--</td>
+                <td class="text-left">Next Due</td>
+                <td class="text-left">{{  items.0.line_items.0.net_amount/100 }}</td>
+              </tr>
+              {% else %}
+                {% for item in items %}
+                  <tr>
+                    <td class="text-left">{{ item.billing_start | date('M d, Y') | default("--") }}</td>
+                    <td class="text-left">{{ item.id }}</td>
+                    <td class="text-left">{{  item.line_items.0.net_amount/100 }}
```
Comment (sonyy-m) [inline catalog/view/theme/default/template/extension/payment/razorpay_subscription_info.twig:152]: add invoice link to the id

Comment (HancilSequeira) [reply catalog/view/theme/default/template/extension/payment/razorpay_subscription_info.twig:152]: @sonyy-m Invoice details are not required for now so I have not add the link for the same

**Pull request merged**